source/robot_model/include/robot_model/Model.hpp

```diff
@@ -185,9 +202,28 @@ class Model {
                                         const state_representation::JointPositions& joint_positions,
                                         const std::vector<std::string>& frames);
 
+
+  /**
+   * @brief Generates a list of collision pairs to exclude based on the kinematic tree of the model
+   * @return the list of collision pairs to exclude 
+  */
+  std::vector<pinocchio::CollisionPair> generate_joint_exclusion_list();
+
 public:
+  /**
+   * @brief Constructor with robot name and path to URDF file and callback function to resolve package paths
+   * @param robot_name the name of the robot
+   * @param urdf_path the path to the URDF file
+   * @param meshloader_callback callback function to resolve package paths
+   */
+  explicit Model(const std::string& robot_name, 
+                   const std::string& urdf_path,
+                   const std::optional<std::function<std::string(const std::string&)>>& meshloader_callback);
+
   /**
    * @brief Constructor with robot name and path to URDF file
+   * @param robot_name the name of the robot
+   * @param urdf_path the path to the URDF file
    */
   explicit Model(const std::string& robot_name, const std::string& urdf_path);
```

Comment:
Some last nitpicks. Each constructor variant now either loads geometries or not, so you should make that explicitly clear in the documentation. I would also keep the "simple" constructor as the first one to be declared. In this way it will also show up first in the documentation. You can use the `@details` tag to give more details, especially to describe the behavior of the meshloader callback.

```suggestion
  /**
   * @brief Construct with robot name and path to URDF file
   * @details If the URDF contains references to collision geometry meshes, they will not be loaded into memory.
   * To enable collision detection, use the alternate constructor.
   * @param robot_name the name to associate with the model
   * @param urdf_path the path to the URDF file
   */
  explicit Model(const std::string& robot_name, const std::string& urdf_path);

  /**
   * @brief Construct a robot model with collision geometries from a URDF file
   * @details If the URDF contains references to collision geometry meshes, they will be loaded into memory.
   * Subsequently, the check_collision() method can be used to check for self-collisions in the robot model.
   * If geometry meshes are referenced with a relative package path using the `package://` prefix, then
   * the optional meshloader_callback function should be defined to return an absolute path to a package
   * given the package name.
   * @param robot_name the name to associate with the model
   * @param urdf_path the path to the URDF file
   * @param meshloader_callback optional callback to resolve the absolute package path from a package name
   */
  explicit Model(const std::string& robot_name, 
                   const std::string& urdf_path,
                   const std::optional<std::function<std::string(const std::string&)>>& meshloader_callback);
```